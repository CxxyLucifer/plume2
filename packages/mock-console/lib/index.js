"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var MockConsole = /** @class */ (function () {
    function MockConsole() {
        this._logs = [];
        this.mock();
    }
    MockConsole.prototype.mock = function () { };
    MockConsole.prototype.mockFn = function () {
        var _this = this;
        return jest.fn(function (log) { return _this._logs.push(log); });
    };
    Object.defineProperty(MockConsole.prototype, "logs", {
        get: function () {
            return this._logs;
        },
        enumerable: true,
        configurable: true
    });
    MockConsole.prototype.reset = function () {
        this._logs = [];
    };
    return MockConsole;
}());
exports.MockConsole = MockConsole;
var MockWarn = /** @class */ (function (_super) {
    __extends(MockWarn, _super);
    function MockWarn() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockWarn.prototype.mock = function () {
        console.warn = this.mockFn();
    };
    return MockWarn;
}(MockConsole));
exports.MockWarn = MockWarn;
var MockLog = /** @class */ (function (_super) {
    __extends(MockLog, _super);
    function MockLog() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockLog.prototype.mock = function () {
        console.log = this.mockFn();
    };
    return MockLog;
}(MockConsole));
exports.MockLog = MockLog;
