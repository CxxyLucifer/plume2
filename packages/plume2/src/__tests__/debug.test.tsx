import { MockLog } from 'mock-console';
import React from 'react';
import renderer from 'react-test-renderer';
import { Actor, Store, StoreProvider } from '../index';

class LoadingActor extends Actor {
  defaultState() {
    return {
      loading: true
    };
  }
}

class HelloActor extends Actor {
  defaultState() {
    return {
      hello: 'hello'
    };
  }
}

class AppStore extends Store {
  bindActor() {
    return [new LoadingActor(), new HelloActor()];
  }
}

@StoreProvider(AppStore, { debug: true })
class DebugApp extends React.Component {
  render() {
    return <div>hello world</div>;
  }
}

it('test debug window app', () => {
  const mockLogs = new MockLog();
  const tree = renderer.create(<DebugApp />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(window['_plume2App'].DebugApp != null).toEqual(true);
  expect(window['_plume2App'].DebugApp.store instanceof AppStore).toEqual(true);
  expect(mockLogs.logs).toMatchSnapshot();
});
