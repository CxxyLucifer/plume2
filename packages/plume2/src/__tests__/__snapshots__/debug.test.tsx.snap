// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test debug window app 1`] = `
<div>
  hello world
</div>
`;

exports[`test debug window app 2`] = `
Array [
  "===========store init==========",
  "Actor:reduce LoadingActor",
  "Actor:reduce HelloActor",
  "%cplume2@1.1.1ðŸš€",
  "StoreProvider(DebugApp) will mount ðŸš€",
]
`;
